tensornetwork/block_sparse/block_tensor_test.py

```diff
@@ -223,3 +224,86 @@ def test_find_diagonal_sparse_blocks(num_legs, num_charges):
 
   assert np.sum(np.prod(ss, axis=0)) == np.sum([len(b) for b in bs])
   np.testing.assert_allclose(unique_left, cs.charges)
+
+
+orders = []
+Ds = []
+for dim, nl in zip([60, 30, 20], [2, 3, 4]):
+  o = list(itertools.permutations(np.arange(nl)))
+  orders.extend(o)
+  Ds.extend([dim] * len(o))
+
+
+@pytest.mark.parametrize('order,D', zip(orders, Ds))
+@pytest.mark.parametrize('num_charges', [1, 2, 3])
+def test_find_transposed_diagonal_sparse_blocks(num_charges, order, D):
+  order = list(order)
+  num_legs = len(order)
+  np.random.seed(10)
+  np_charges = [
+      np.random.randint(-5, 5, (num_charges, D), dtype=np.int16)
+      for _ in range(num_legs)
+  ]
+  tr_fused = np.stack([
+      fuse_ndarrays([np_charges[order[n]][c, :]
+                     for n in range(num_legs)])
+      for c in range(num_charges)
+  ],
+                      axis=0)
+  fused = np.stack([
+      fuse_ndarrays([np_charges[n][c, :]
+                     for n in range(num_legs)])
+      for c in range(num_charges)
+  ],
+                   axis=0)
+
```

Comment:
These are really hard to parse. Remember, there's no same in using multiple lines of code!

As a general rule of thumb, you generally want to avoid nested list comprehension, especially when it has to be broken into multiple lines like this. (We're explicitly not allowed to use it in Google's codebase). Prefer to use a normal for-loop for situations like this.

Reply:
sound goods, I'll change it!